cub/cub/block/specializations/block_reduce_raking.cuh

```diff
@@ -225,6 +225,9 @@ struct BlockReduceRaking
 
         int valid_raking_threads = (IS_FULL_TILE) ? RAKING_THREADS : (num_valid + SEGMENT_LENGTH - 1) / SEGMENT_LENGTH;
 
+        // sync before re-using shmem (warp_storage/raking_grid are aliased)
+        CTA_SYNC();
```

Comment:
Without knowing what the algorithm does, the comment above says "Reduce parallelism to one warp" and later we sync the whole block. Is this legal? I thought the entire block needs to execute a `CTA_SYNC()`.

Reply:
Good catch, this is ill-formed. I'll revise.